pkg/controller/clusterroleaggregation/clusterroleaggregation_controller.go

```diff
@@ -121,17 +122,43 @@ func (c *ClusterRoleAggregationController) syncClusterRole(key string) error {
 		return nil
 	}
 
-	// we need to update
-	clusterRole := sharedClusterRole.DeepCopy()
-	clusterRole.Rules = nil
-	for _, rule := range newPolicyRules {
-		clusterRole.Rules = append(clusterRole.Rules, *rule.DeepCopy())
-	}
-	_, err = c.clusterRoleClient.ClusterRoles().Update(context.TODO(), clusterRole, metav1.UpdateOptions{})
+	clusterRole := rbacv1ac.ClusterRole(sharedClusterRole.Name).
+		SetRules(toApplyPolicyRules(newPolicyRules)...)
+
+	opts := metav1.ApplyOptions{FieldManager: "clusteraggregationcontroller", Force: true}
```

Comment:
Do we have a convention for the field manager name? (If it's based on file name I guess it should be clusterroleaggregationcontroller). It probably shouldn't matter since we're using force, but just want to double check

Edit: It seems that the current field manager name is based on the user agent "clusterrole-aggregation-controller" (https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/rbac.go#L32).

Reply:
Thanks for doing the lookup. We should probably use "clusterrole-aggregation-controller" here. I.e. start a convention where the default fieldmanager name to use is the user agent, and to only deviate from that if you need multiple field managers. I'll update.